### PR TITLE
RavenDB-18619_2 - IncludeExplanations doesn't work correctly

### PR DESCRIPTION
**Pull request by shaharhikri**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18619

### Additional description

IncludeExplanations doesn't work correctly.

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Not relevant

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Review by grisha-kotler — APPROVED** [review]

File: src/Raven.Client/Documents/Queries/Explanation/Explanations.cs [hunk 1]
```diff
@@ -6,6 +6,8 @@ public class Explanations
     {
         private Dictionary<string, string[]> _explanations;
 
+        public bool ShouldBeIncluded { get; set; } = false;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Queries/Explanation/Explanations.cs]: in QueryTimings we made it internal (#14048). Why here it should be public? Also isn't false the default value for bool?

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Explanations.cs [hunk 2]
```diff
@@ -17,7 +17,9 @@ public void IncludeExplanations(ExplanationOptions options, out Explanations exp
 
             var optionsParameterName = options != null ? AddQueryParameter(options) : null;
             ExplanationToken = ExplanationToken.Create(optionsParameterName);
```
Comment (grisha-kotler) [inline src/Raven.Client/Documents/Session/AbstractDocumentQuery.Explanations.cs:19]: Why do we have an `ExplanationToken` and `Explanations`?
Can we have a single object?

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Session/AbstractDocumentQuery.Explanations.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.Explanations.cs:19]: Is there a good advantage of changing the implementation at this point?

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/AbstractDocumentQuery.Explanations.cs:19]: Probably not.

File: src/Raven.Client/Documents/Queries/Explanation/Explanations.cs (hunk 1 above)
Comment (shaharhikri) [reply src/Raven.Client/Documents/Queries/Explanation/Explanations.cs]: Done

**Pull request merged**